src/drive.ts

```diff
@@ -230,9 +225,15 @@ async function getFileList(
         url += '&format=' + format;
     }
 
-    return await requestAPI('/api/contents/' + path + '' + url, {
-        method: 'get'
-    });
+    const result: Contents.IModel  = await requestAPI('/api/contents/' + path + '' + url, {method: 'get'});
+    if (!showHidden && Array.isArray(result.content)) {
+        const hiddenFilesNo: number = result.content.filter((file: { name: string; }) => file.name.startsWith('.')).length
+        await stateDB.save('hiddenFilesNo', hiddenFilesNo)
+        const filteredResult = JSON.parse(JSON.stringify(result))
+        filteredResult.content = (result.content as Array<any>).filter((file: { name: string; }) => !file.name.startsWith('.'))
```

Comment:
I mention that maybe we should not do any request to backend when changing between show/hide. So, can't we store this somehow and filter only when printing in the UI?

Reply:
I simplified it, now there are no additional calls to backend, the information about hidden files number is fetched during file list request

Reply:
But we still get to the backend every time we switch between show/hide the hidden files, right?

Reply:
No, Jupyter pings /contents endpoint every several seconds by default and pings this endpoint as well during directory change or file list refresh, I use result of this request to count current number of files. When switching between show/hide no additional request is made, it's just for label change.

Reply:
ok!